### DISCUSSION TITLE
Working directory to be dynamic, like a note's current path

### DISCUSSION THREAD

**Discussion by m00nlygreat** [Taitava/obsidian-shellcommands#437] [Ideas]
it seems my command needs to work in the directory where my current note is located, but I failed to find that option in the settings

**Comment by Taitava** [comment]
Hi,

yeah, the [Working directory](https://publish.obsidian.md/shellcommands/Basic+usage/Working+directory) setting does not currently support variables. If it did, then it would be possible to use e.g. [`{{folder_path:absolute}}`](https://publish.obsidian.md/shellcommands/Variables/%7B%7Bfolder_path%7D%7D) as a working directory.

Then again, this kind of improvement would not be strictly necessary, as you can just prefix your shell command with `cd {{folder_path:relative}}` or `cd {{folder_path:absolute}}` (both would do the same thing).

An example command that outputs all files and folders located in the currently open note's folder:
```bash
cd {{folder_path:absolute}}
ls
```
